packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -595,32 +598,19 @@ async function setupOrgBrowser(
 }
 
 export async function activate(extensionContext: vscode.ExtensionContext) {
-  const rootWorkspacePath = getRootWorkspacePath();
-  if (
-    rootWorkspacePath &&
-    process.cwd() !== rootWorkspacePath &&
-    fs.existsSync(rootWorkspacePath)
-  ) {
-    try {
-      // Switch to the project directory so that the main @salesforce
-      // node libraries work correctly.  @salesforce/core,
-      // @salesforce/source-tracking, etc. all use process.cwd()
-      // internally.  This causes issues when used from VSCE, as VSCE
-      // processes can run with a path that does not reflect the current
-      // project path (it often returns '/' from process.cwd()).
-      // Switching to the project path here at activation time ensures that
-      // commands are run with the project path returned from process.cwd(),
-      // thus avoiding the potential errors surfaced when the libs call
-      // process.cwd().
-      process.chdir(rootWorkspacePath);
-    } catch (error) {
-      console.log(
-        'There was an error switching to the root project path on activation.'
-      );
-    }
-  }
-
   const extensionHRStart = process.hrtime();
+  const rootWorkspacePath = getRootWorkspacePath();
+  // Switch to the project directory so that the main @salesforce
+  // node libraries work correctly.  @salesforce/core,
+  // @salesforce/source-tracking, etc. all use process.cwd()
+  // internally.  This causes issues when used from VSCE, as VSCE
+  // processes can run with a path that does not reflect the current
+  // project path (it often returns '/' from process.cwd()).
+  // Switching to the project path here at activation time ensures that
+  // commands are run with the project path returned from process.cwd(),
+  // thus avoiding the potential errors surfaced when the libs call
+  // process.cwd().
+  ensureCurrentWorkingDirIsProjectPath(rootWorkspacePath);
```

Comment:
I decided not to wrap this in a try/catch since downstream functions will all be depending on cwd() returning the projectPath from this point forward.  If something goes wrong with switching to the projectPath, then I think activate should throw here.